jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -123,7 +131,10 @@ object XGBoost extends Serializable {
           }
           val partitionItr = fromDenseToSparseLabeledPoints(trainingSamples, missing)
           val trainingSet = new DMatrix(new JDMatrix(partitionItr, cacheFileName))
-          booster = SXGBoost.train(trainingSet, xgBoostConfMap, round,
+          if (groupData != null) {
+            trainingSet.setGroup(groupData(partIndex).toArray)
+          }
```

Comment:
Sorry, I can not understand the intent of the question correctly.
If my answer dose not make sense, please let me know.

I want to call [ setGroup(group: Array[Int])](https://github.com/dmlc/xgboost/blob/37bc122c90b7c1337403f797fb7b01fc6abd5b36/jvm-packages/xgboost4j/src/main/scala/ml/dmlc/xgboost4j/scala/DMatrix.scala#L148)

And [checking xgBoostConfMap](https://github.com/cloverrose/xgboost/blob/aac2edeba51ee47df6030c35b6c0da135c4051d9/jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala#L111-L114) is costly.

So I check xgBoostConfMap and make independ variable.